atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1700,28 +1703,33 @@ private void commitWrites(TransactionService transactionService) {
                 // We check the pre-commit conditions first since they may operate similarly to read write conflict
                 // handling - we should check lock validity last to ensure that sweep hasn't affected the checks.
                 timedAndTraced("userPreCommitCondition", () -> throwIfPreCommitConditionInvalid(commitTimestamp));
-                timedAndTraced("preCommitLockCheck", () -> throwIfImmutableTsOrCommitLocksExpired(commitLocksToken));
+                traced("preCommitLockCheck", () -> throwIfImmutableTsOrCommitLocksExpired(commitLocksToken));
 
-                timedAndTraced("commitPutCommitTs",
+                // TransactionService.putUnlessExists: actually commits the transaction
```

Comment:
I think this comment, in isolation, is not that useful: I assume it is explaining why this is not timed, but outside of the context of this PR it doesn't really imply. Perhaps:

```// Not timed because TransactionService.putUnlessExists actually commits the transaction, and is timed```

Reply:
That's a fair point. Makes sense